### PR TITLE
[MINOR][SQL] Some errors in the notes.

### PR DESCRIPTION
**Pull request by CarolinePeng**
## What changes were proposed in this pull request?

When using ordinals to access linked list, the time cost is O(n).

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
It's okay but mind taking another look and fix some more typos together while we're here? I'm pretty sure there are more.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala [hunk 1]
```diff
@@ -101,7 +101,7 @@ package object expressions  {
       StructType(attrs.map(a => StructField(a.name, a.dataType, a.nullable, a.metadata)))
     }
 
-    // It's possible that `attrs` is a linked list, which can lead to bad O(n^2) loops when
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala]: Could you tell us why you think this comment is error? This comments came from [SPARK-15764 Replace N^2 loop in BindReferences](https://github.com/apache/spark/pull/13505) to explain the problamatic situation.

Comment (CarolinePeng) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala]: The cost time to access the linked list using ordinal is O (n), but I do not combine BindReferences to analyze the cost time.

Comment (dongjoon-hyun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala]: Ya. The original sentence has a different context at that time.

**Comment by srowen** [comment]
The merge error is spurious; merging to master

**Pull request closed without merging**